libraries/botbuilder-dialogs-declarative/src/resources/resourceExplorer.ts

```diff
@@ -240,9 +240,13 @@ export class ResourceExplorer {
         );
     }
 
-    private registerTypeInternal(kind: string, type: new () => {}, loader?: CustomDeserializer): void {
+    private registerTypeInternal(
```

Comment:
Maybe this would provide some more flexibility:

```typescript
private registerTypeInternal<T, C>(kind: string, type: new (...args: unknown[]): T, loader?: CustomerDeserializer<T, C> {
  ...
})
```